OpenRA.Game/FileSystem/ZipFile.cs

```diff
@@ -206,7 +206,11 @@ public Stream GetSource()
 
 		public bool TryParsePackage(Stream s, string filename, FileSystem context, out IReadOnlyPackage package)
 		{
-			if (!Extensions.Any(e => filename.EndsWith(e, StringComparison.InvariantCultureIgnoreCase)))
+			var signature = new byte[4];
+			s.ReadBytes(signature, 0, 4);
```

Comment:
`var signature = s.ReadUInt32();` and then `signature != 0x04034b50` would be simpler.